### PR TITLE
Single softmax

### LINKED ISSUE #112: Have one softmax_logits function
# Feature
## Desired Behavior / Functionality
Instead of having multiple instances of the function softmax_logits in different files, have one single one in a utils.py file or similar

## What Needs to Be Done
Move the softmax function into a shared file. Remove all other instances of it and import it instead.

## How Can It Be Tested
-

### PR DESCRIPTION
**Pull request by johanos1**
# Description

Summary of changes

* Moved softmax_logits to mia_attacks/utils/utils.py
* Updated RMIA to use this function
* Updated boosting to use this function.

## Resolved Issues

- [ ] fixes #112 

## How Has This Been Tested?
- The new softmax computation has been tested against the previous in every place where it was replaced.
- Unit test of utils.py

# Related PRs
- #83

### CODE REVIEW

File: leakpro/attacks/utils/boosting.py [hunk 1]
```diff
@@ -119,7 +119,7 @@ def _memorization_score(self:Self) -> None:
         logits_function = ModelLogits()
         logits = np.swapaxes(logits_function(self.shadow_models, self.handler, self.audit_data_indices, self.batch_size), 0, 1)
 
-        logits = self.softmax_logits(logits)
+        logits = softmax_logits(logits)
```
Comment (henrikfo) [inline leakpro/attacks/utils/boosting.py:122]: Either change the default value for temp to 1 or set it to one here

File: leakpro/attacks/utils/boosting.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ def _memorization_score(self:Self) -> None:
             #       filtering of IN- vs. OUT-samples.
             target_logits = np.swapaxes(logits_function([self.target_model], self.handler, self.audit_data_indices,\
                                         self.batch_size), 0, 1).squeeze()
-            target_logits = self.softmax_logits(target_logits)
+            target_logits = softmax_logits(target_logits)
```
Comment (henrikfo) [inline leakpro/attacks/utils/boosting.py:139]: Since you seems to be be setting the temp for all softmax uses in RMIA, I suggest to change the default value for temp to 1 or set it to one here

File: leakpro/attacks/utils/utils.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
+"""Utility functions for attacks."""
+
+import numpy as np
+from torch import exp, from_numpy, max, sum
+
+
+def softmax_logits(logits: np.ndarray, temp:float=2.0) -> np.ndarray:
```
Comment (henrikfo) [inline leakpro/attacks/utils/utils.py:7]: Maybe add dimension as an input variable, default -1?

File: leakpro/attacks/utils/boosting.py (hunk 2 above)
Comment (johanos1) [reply leakpro/attacks/utils/boosting.py:139]: done.

File: leakpro/attacks/utils/utils.py (hunk 3 above)
Comment (johanos1) [reply leakpro/attacks/utils/utils.py:7]: done.

**Review by henrikfo — APPROVED** [review]
